src/ripple/protocol/impl/BuildInfo.cpp

```diff
@@ -77,82 +77,88 @@ getFullVersionString()
+                                          std::string_view prefix,
+                                          std::uint8_t key,
+                                          std::uint8_t lok,
+                                          std::uint8_t hik) -> std::uint8_t {
+                    std::uint8_t ret = 0;
+
+                    if (prefix != identifier.substr(0, prefix.length()))
+                        return 0;
+
+                    if (!beast::lexicalCastChecked(
+                            ret,
+                            std::string(identifier.substr(prefix.length()))))
+                        return 0;
+
+                    if (std::clamp(ret, lok, hik) != ret)
+                        return 0;
 
-                for (auto id : v.preReleaseIdentifiers)
+                    return ret + key;
+                };
+
+                x = parsePreRelease(id, "rc", 0x80, 0, 63);
+
+                if (x == 0)
+                    x = parsePreRelease(id, "b", 0x40, 0, 63);
+
+                if (x & 0xC0)
                 {
-                    auto parsePreRelease =
-                        [](std::string_view identifier,
-                           std::string_view prefix,
-                           std::uint8_t key,
-                           std::uint8_t lok,
-                           std::uint8_t hik) -> std::uint8_t {
-                        std::uint8_t ret = 0;
-
-                        if (prefix != identifier.substr(0, prefix.length()))
-                            return 0;
-
-                        if (!beast::lexicalCastChecked(
-                                ret,
-                                std::string(
-                                    identifier.substr(prefix.length()))))
-                            return 0;
-
-                        if (std::clamp(ret, lok, hik) != ret)
-                            return 0;
-
-                        return ret + key;
-                    };
-
-                    x = parsePreRelease(id, "rc", 0x80, 0, 63);
-
-                    if (x == 0)
-                        x = parsePreRelease(id, "b", 0x40, 0, 63);
-
-                    if (x & 0xC0)
-                    {
-                        c |= static_cast<std::uint64_t>(x) << 16;
-                        break;
-                    }
+                    c |= static_cast<std::uint64_t>(x) << 16;
+                    break;
                 }
             }
         }
+    }
 
-        return c;
-    }();
+    return c;
+}
 
+std::uint64_t
+getEncodedVersion()
+{
+    static std::uint64_t const cookie = {encodeSoftwareVersion(versionString)};
     return cookie;
 }
 
 bool
 isNewerVersion(std::uint64_t version)
 {
-    std::uint64_t constexpr mask = 0x183B000000000000;
-    if ((version & mask) == mask)
+    if ((version & implementationVersionIdentifierMask) ==
+        implementationVersionIdentifier)
         return version > getEncodedVersion();
     return false;
```

Comment:
FWIW, this `return false` is not exercised by the unit tests.

Reply:
Added a test case. I realized I should test this case right after I pushed last time.